compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -235,14 +234,27 @@ bitflags::bitflags! {
 
 pub const DELEGATION_INHERIT_ATTRS_START: DelegationFnSigAttrs = DelegationFnSigAttrs::MUST_USE;
 
+#[derive(Debug)]
+pub struct DelegationInfo {
+    // NodeId (either delegation.id or item_id in case of a trait impl) for signature resolution,
+    // for details see https://github.com/rust-lang/rust/issues/118212#issuecomment-2160686914
+    pub resolution_node: ast::NodeId,
+    pub attributes: DelegationAttributes,
+}
+
+#[derive(Debug)]
+pub struct DelegationAttributes {
+    pub flags: DelegationFnSigAttrs,
+    pub to_inherit_attrs: AttrVec,
+}
+
 #[derive(Debug)]
 pub struct DelegationFnSig {
     pub header: ast::FnHeader,
     pub param_count: usize,
     pub has_self: bool,
     pub c_variadic: bool,
-    pub attrs_flags: DelegationFnSigAttrs,
-    pub to_inherit_attrs: AttrVec,
+    pub attributes: DelegationAttributes,
```

Comment:
```suggestion
    pub attrs: DelegationAttrs,
```
The word "attributes" is long and is typically abbreviated in the compiler.